clang/lib/Driver/ToolChains/BareMetal.cpp

```diff
@@ -336,6 +336,26 @@ BareMetal::OrderedMultilibs BareMetal::getOrderedMultilibs() const {
   return llvm::reverse(Default);
 }
 
+ToolChain::CXXStdlibType BareMetal::GetDefaultCXXStdlibType() const {
+  if (getTriple().isRISCV() && GCCInstallation.isValid())
```

Comment:
Could we make it conditional only on valid GCC installation?
```suggestion
  if (GCCInstallation.isValid())
```

Reply:
This will modify the behavior for other targets. I understand that other targets can benefit from this as well however in this series of patches I will like to preserve the behavior as it was before merging RISCVToolchain into Baremetal toolchain object. If needed, I can push a separate PR (not a part of this series of patches) after merging of toolchain objects is done to ensure that other downstream toolchain vendors are not affected by this series of patches as it will already cause some churn in their code.

I will like to keep it as is until there is strong opinion to change the behavior in this patch itself